dbms/src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -306,9 +306,17 @@ class AggregateFunctionSum final : public IAggregateFunctionDataHelper<Data, Agg
 
     AggregateFunctionSum() = default;
 
-    AggregateFunctionSum(PrecType prec, ScaleType scale)
+    AggregateFunctionSum(PrecType prec, ScaleType scale, bool is_final = false)
     {
-        std::tie(result_prec, result_scale) = SumDecimalInferer::infer(prec, scale);
+        if (is_final)
```

Comment:
How about referring to `countSecondStage`?
```
struct NameFinalSum
{
    static constexpr auto name = "finalSum";
};

template <typename T>
using AggregateFunctionFinalSum = AggregateFunctionSum<T, T, AggregateFunctionSumData<T>, NameFinalSum>;

AggregateFunctionSum(PrecType prec, ScaleType scale)
{
    if constexpr (Name::name == NameFinalSum::name)
    {
            result_prec = prec;
            result_scale = scale;
    }
    else
    {
        std::tie(result_prec, result_scale) = SumDecimalInferer::infer(prec, scale);
    }
}
```

Reply:
I think we need to make a clear distinction between tipb's code and clickhouse's code.

Reply:
> 

sure